scripts/tasks/check.ts

```diff
@@ -12,12 +12,13 @@ const nolinkCommand = `npx nx run-many -t check -c production ${parallel}`;
 
 export const check: Task = {
   description: 'Typecheck the source code of the monorepo',
+  dependsOn: ['compile'],
```

Comment:
I think it's preferable if we can still typecheck our `src` without compiling first; this would make that impossible?

Reply:
All of the `nx` `check` tasks already depend on the `build` task (which is, confusingly, what the compile task runs...), so this makes minimal difference; at the very least it should include `install` to support running type checking from a cold-start

Reply:
I suppose it does somewhat reduce the degree of parallelisation available as all build tasks need to be completed before check tasks can start. The stacked task runners do make this hard to follow sometimes.